### PR TITLE
Added Item, ItemFactory, and ItemFactoryTest class

### PR DESCRIPTION
**Pull request by wang-xiaotian**
Updated Item, ItemFactory, ItemFactoryTest and setItems(List) in Inventory class

### CODE REVIEW

File: src/main/java/com/zombiecastlerush/building/Inventory.java [hunk 1]
```diff
@@ -12,6 +12,10 @@
     public Inventory(){
     }
 
+    public void setItems(List<Item> list){
```
Comment (chaynsaw) [inline src/main/java/com/zombiecastlerush/building/Inventory.java:27]: - [ ] Make sure to add this to the [Game Mock-up](https://docs.google.com/presentation/d/14OPQWXwFgpWjsf6KbH2bE3556LCGIcyVj94nyy2Md6g/edit#slide=id.gdefcc1da4f_1_26)

File: src/test/java/com/zombiecastlerush/building/ItemFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            Item item = new Item("item"+i, "i am #"+i + " item");
+            items.add(item);
+        }
+        ItemFactory.writeItemsToDir(file, items); // generate .json file
+    }
+
+    @Test
+    public void testReadItemsFromDirectory_returnAListOfItems() throws IOException {
+        Inventory inventory = new Inventory();
+        File file = new File(outputPath);
```
Comment (chaynsaw) [inline src/test/java/com/zombiecastlerush/building/ItemFactoryTest.java:34]: Does this need to be done in both setUp() and again in this method?

Comment (wang-xiaotian) [reply src/test/java/com/zombiecastlerush/building/ItemFactoryTest.java:34]: Will test and fix

File: src/test/java/com/zombiecastlerush/building/ItemFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        List<Item> list = ItemFactory.readItemsFromDir(inventory, file);
+        for(int i = 0; i < list.size(); i++){
+            // expectedItem is how the Item class implements toString(), but manually made; needs to change accordingly
+            String expectedItem = String.format("name= item%d description= i am #%d item",i, i);
+            System.out.println(list.get(i).toString());
+            System.out.println(expectedItem);
+           // Assert.assertTrue(list.get(i).toString().equals(expectedItem));
```
Comment (wang-xiaotian) [inline src/test/java/com/zombiecastlerush/building/ItemFactoryTest.java]: remove line 41

**Review by chaynsaw — APPROVED** [review]

File: src/test/java/com/zombiecastlerush/building/ItemFactoryTest.java (hunk 2 above)
Comment (wang-xiaotian) [reply src/test/java/com/zombiecastlerush/building/ItemFactoryTest.java:34]: Removed line 41

**Review by gtulkn12 — APPROVED** [review]
Ready to merge

**Pull request merged**